app/src/main/java/com/example/shiv/mdbpokedexproject2/FilterActivity.java

```diff
@@ -0,0 +1,387 @@
+        ToggleButton normal = (ToggleButton) findViewById(R.id.toggleButtonNormal);
+        ToggleButton fire = (ToggleButton) findViewById(R.id.toggleButtonFire);
+        ToggleButton water = (ToggleButton) findViewById(R.id.toggleButtonWater);
+        ToggleButton grass = (ToggleButton) findViewById(R.id.toggleButtonGrass);
+        ToggleButton electric = (ToggleButton) findViewById(R.id.toggleButtonElectric);
+        ToggleButton ice = (ToggleButton) findViewById(R.id.toggleButtonIce);
+        ToggleButton ground = (ToggleButton) findViewById(R.id.toggleButtonGround);
+        ToggleButton flying = (ToggleButton) findViewById(R.id.toggleButtonFlying);
+        ToggleButton poison = (ToggleButton) findViewById(R.id.toggleButtonPoison);
+        ToggleButton fighting = (ToggleButton) findViewById(R.id.toggleButtonFighting);
+        ToggleButton psychic = (ToggleButton) findViewById(R.id.toggleButtonPsychic);
+        ToggleButton dark = (ToggleButton) findViewById(R.id.toggleButtonDark);
+        ToggleButton rock = (ToggleButton) findViewById(R.id.toggleButtonRock);
+        ToggleButton bug = (ToggleButton) findViewById(R.id.toggleButtonBug);
+        ToggleButton ghost = (ToggleButton) findViewById(R.id.toggleButtonGhost);
+        ToggleButton steel = (ToggleButton) findViewById(R.id.toggleButtonSteel);
+        ToggleButton allTypes = (ToggleButton) findViewById(R.id.toggleButtonAllTypes);
+
+
+
+
+
+
+
+        min_Attack.setText("" + MainActivity.min_attack);
+        min_Defense.setText("" + MainActivity.min_defense);
+        min_health.setText("" + MainActivity.min_health);
+
+
+
+        button2.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                try {
+                    MainActivity.min_attack = Integer.parseInt(min_Attack.getText().toString());
+                } catch (Throwable e) {
+                    MainActivity.min_attack = 0;
+                }
+                try {
+                    MainActivity.min_defense = Integer.parseInt(min_Defense.getText().toString());
+                } catch (Throwable e) {
+                    MainActivity.min_defense = 0;
+                }
+                try {
+                    MainActivity.min_health = Integer.parseInt(min_health.getText().toString());
+                } catch (Throwable e) {
+                    MainActivity.min_health = 0;
+                }
+
+                Intent intent = new Intent(getApplicationContext(), MainActivity.class);
+
+                Toast toast = Toast.makeText(getApplicationContext(), "Filter Updated!", Toast.LENGTH_SHORT);
+                toast.show();
+
+                startActivity(intent);
+            }
+        });
+
+        MainActivity.displayAllTypes = false;
+        //Setting up all toggle buttons
+        dragon.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
```

Comment:
Creating that private method wouldn't clean up the code by much since I would have to call that method the same number of times